ddtrace/contrib/asgi/middleware.py

```diff
@@ -183,4 +189,5 @@ async def wrapped_send(message):
                 del scope["datadog"]["request_span"]
             except KeyError:
                 pass
```

Comment:
Marking this resolved because the `span.finish()` in the `middleware.py` will still be there with this branch:

```
        try:
            return await self.app(scope, receive, wrapped_send)
        except Exception as exc:
            (exc_type, exc_val, exc_tb) = sys.exc_info()
            span.set_exc_info(exc_type, exc_val, exc_tb)
            self.handle_exception_span(exc, span)
            reraise(exc_type, exc_val, exc_tb)
        finally:
            try:
                del scope["datadog"]["request_span"]
            except KeyError:
                pass

            span.finish()
```

See https://github.com/DataDog/dd-trace-py/pull/3799#discussion_r893903331 . I was worried about calling `span.finish()` multiple times, but it looks like only the first `finish` is used in the duration. 

The `finally` portion seems to always have access to the `span`. 

If a span is finished after `more_data` is false, then that shouldn't impact the `span.finish()` in the finally section.

Reply:
yup, I am dense and didn't look at the next (hidden) line in this file. I should have done that.

expectations are already met here.